### DISCUSSION TITLE
ScriptReference/MonoBehaviour.Reset

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#19] [Notes]
# ScriptReference/MonoBehaviour.Reset

https://docs.unity3d.com/ScriptReference/MonoBehaviour.Reset.html

**Comment by vertxxyz** [comment]
This function is called before deserialization. This means that when it gets called after a **Paste Component As New** operation you both cannot operate on serialized data you expect to be on the object, nor can you expect your default values to persist in this case.
